src/common/exif.cc

```diff
@@ -3988,8 +4005,31 @@ int dt_exif_xmp_attach_export(const int imgid, const char *filename, void *metad
   }
   catch(Exiv2::AnyError &e)
   {
-    std::cerr << "[dt_exif_xmp_attach_export] " << filename << ": caught exiv2 exception '" << e << "'\n";
-    return -1;
+    std::string s(e.what());
+    if(s == "Size of XMP JPEG segment is larger than 65535 bytes")
+    {
+      Exiv2::XmpData &xmpData = img->xmpData();
+      _remove_xmp_keys(xmpData, "Xmp.darktable.history");
+      _remove_xmp_keys(xmpData, "Xmp.darktable.masks_history");
+      _remove_xmp_keys(xmpData, "Xmp.darktable.auto_presets_applied");
+      _remove_xmp_keys(xmpData, "Xmp.darktable.iop_order");
+
+      try
+      {
+        img->writeMetadata();
```

Comment:
So you're assuming the exception was thrown by img->writeMetadata() above. Then add try-catch block around that call, check if error code is what you want and try to handle it. Re-throw (just plain throw) otherwise. This way you don't have to move img outside of try block or re-getting xmpData.

Reply:
> So you're assuming the exception was thrown by img->writeMetadata() above.

I've checked this. The oversize exception is generated there.

Reply:
>  Then add try-catch block around that call, check if error code is what you want and try to handle it.

Sounds great. I'll make a try.